### ISSUE TITLE
Element ref behavior differs in development and production for initial render with new CSS support

### ISSUE THREAD

**Issue by chemicalkosek** [zeit/next.js#10163]
# Bug report

## Describe the bug

After updating to next@9.2 it seems that the refs are not being created during server-side render in development mode. Everything works fine after building to production, also everything works fine when mounting that component client-side. Also, everything works fine pre 9.2. Disclaimer: I may be wrong with my conclusion of what is really the cause of the bug. It may as well be with how I am rendering those components.

## To Reproduce

My code: (this is yet to be refactored to hooks).
```
class ScheduleComponent extends React.Component {
  schedule = React.createRef();

  state = {
    visibility: 'hidden',
    width: 1240,
    leftBond: moment()
      .subtract(7, 'days')
      .startOf('day'),
    rightBond: moment()
      .add(23, 'days')
      .startOf('day'),
  };

  componentDidMount() {
    this.setState({
      width: this.schedule.current.scrollWidth,
      visibility: 'visible',
    });
    window.addEventListener('resize', this.resize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.resize);
  }

  resize = () => {
    this.setState({ width: this.schedule.current.scrollWidth });
  };
```

## Expected behavior

Refs should be created during server-side render in dev mode.

## Screenshots

What's happening during server-side render. The elements with month names and year should be already rendered. They render only in production or during client-side mounting or after firing the resize function as you can see on the gif.

![Peek 2020-01-19 19-33](https://user-images.githubusercontent.com/11018707/72686226-a69c9e00-3af2-11ea-834d-be77113cd3ac.gif)

## System information

- OS: Ubuntu 18.04.3
- Browser: Firefox, Chrome
- Version of Next.js: 9.2

**Comment by JarrodMFlesch** [comment]
Haha! I was literally just going to create an issue!

Here are two sandbox links to show the reproducible issue.

[Does not work 9.2.0](https://codesandbox.io/s/dom-ref-test-920-6e5yt)
[Does work 9.1.7](https://codesandbox.io/s/dom-ref-test-917-6lbz7)

**Comment by jacobsfletch** [comment]
I experience these same breaking changes when toggling between 9.1.7 to 9.2.0. Server-side refs seem to be absent in dev.

**Comment by ijjk** [comment]
@JarrodMFlesch it looks like the reproduction you showed is due to the new CSS support since in development when the page is first loaded we add a `body{display:none}` style tag to the head which is removed after the styles are applied.

Disabling experimental css with `experimental: { css: false }` in `next.config.js` returns to the previous ref behavior in development mode. We will need to investigate this some more since refs behaving differently between production and development isn't ideal

**Comment by ijjk** [comment]
This should be resolved with #10164 and available in Next.js >= `v9.2.1-canary.4`

**Comment by JarrodMFlesch** [comment]
> This should be resolved with #10164 and available in Next.js >= `v9.2.1-canary.4`

You guys rock.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
